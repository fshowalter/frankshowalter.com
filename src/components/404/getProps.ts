import { booklogUpdates } from "~/api/booklog";
import { getFluidImageProps } from "~/api/images";
import { movielogUpdates } from "~/api/movielog";
import { ImageConfig } from "~/components/LatestUpdates";

import type { Props } from "./404";

export async function getProps({
  deck,
  title,
}: {
  deck: string;
  title: string;
}): Promise<Props> {
  const booklogItems = await booklogUpdates();
  const movielogItems = await movielogUpdates();

  return {
    booklogUpdates: await Promise.all(
      booklogItems.map(async (item) => {
        return {
          ...item,
          imageProps: await getFluidImageProps("cover", item.slug, ImageConfig),
        };
      }),
    ),
    deck,
    movielogUpdates: await Promise.all(
      movielogItems.map(async (item) => {
        return {
          ...item,
          imageProps: await getFluidImageProps(
            "poster",
            item.slug,
            ImageConfig,
          ),
        };
      }),
    ),
    title,
  };
}
