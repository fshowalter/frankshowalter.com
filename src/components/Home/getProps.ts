import { getBackdropImageProps } from "~/api/backdrops";
import { booklogUpdates } from "~/api/booklog";
import { getFluidImageProps } from "~/api/images";
import { movielogUpdates } from "~/api/movielog";
import { BackdropImageConfig } from "~/components/Backdrop";
import { ImageConfig } from "~/components/LatestUpdates";

import type { Props } from "./Home";

export async function getProps(): Promise<Props> {
  const booklogItems = await booklogUpdates();
  const movielogItems = await movielogUpdates();

  return {
    backdropImageProps: await getBackdropImageProps(
      "home",
      BackdropImageConfig,
    ),
    booklogUpdates: await Promise.all(
      booklogItems.map(async (item) => {
        return {
          ...item,
          imageProps: await getFluidImageProps(
            "cover",
            item.slug,
            ImageConfig.width,
          ),
        };
      }),
    ),
    movielogUpdates: await Promise.all(
      movielogItems.map(async (item) => {
        return {
          ...item,
          imageProps: await getFluidImageProps(
            "poster",
            item.slug,
            ImageConfig.width,
          ),
        };
      }),
    ),
  };
}
