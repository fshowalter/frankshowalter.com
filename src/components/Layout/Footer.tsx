import type { JSX } from "react";

export function Footer(): JSX.Element {
  return (
    <div
      className={`
        flex flex-wrap items-start justify-between gap-[10%] bg-footer
        px-container py-10 text-inverse
        tablet:px-12 tablet:pt-10
        desktop:p-10
      `}
    ></div>
  );
}
