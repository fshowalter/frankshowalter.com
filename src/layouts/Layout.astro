---
import "./tailwind.css";

type Meta = {
  description: string;
};

type OpenGraph = {
  description: string;
  type: "article" | "website";
};

type Props = {
  canonical: boolean;
  meta: Meta;
  openGraph: false | OpenGraph;
  title: string;
};

const siteMeta = {
  author: "Frank Showalter",
  title: "Frank Showalter",
};

const { canonical, meta, openGraph, title } = Astro.props;

const pageTitle = title.startsWith(siteMeta.title)
  ? title
  : `${title} - ${siteMeta.title}`;

const canonicalUrl = `${import.meta.env.SITE}${new URL(Astro.request.url).pathname}`;
---

<!doctype html>
<html class="h-full" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta content="width=device-width" name="viewport" />
    {canonical && <link href={canonicalUrl} rel="canonical" />}
    <link href="/favicon.svg" rel="icon" sizes="any" type="image/svg+xml" />
    <link href="/sitemap-index.xml" rel="sitemap" />
    <meta content={meta.description} name="description" />
    <slot name="meta-theme-color">
      <meta
        content="#8f8f8f"
        media="(prefers-color-scheme: light)"
        name="theme-color"
      />
      <meta
        content="#484745"
        media="(prefers-color-scheme: dark)"
        name="theme-color"
      />
    </slot>
    {
      openGraph && (
        <>
          <meta content={`${canonicalUrl}og.jpg`} property="og:image" />
          <meta content={`${canonicalUrl}`} property="og:url" />
          <meta content={title} property="og:title" />
          <meta content={openGraph.description} property="og:description" />
          <meta content={openGraph.type} property="og:type" />
        </>
      )
    }
    <title>{pageTitle}</title>
  </head>
  <body class="h-full">
    <slot />
  </body>
</html>
