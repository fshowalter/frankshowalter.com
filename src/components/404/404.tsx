import type { JSX } from "react";

import { SolidBackdrop } from "~/components/Backdrop";
import { LatestUpdates } from "~/components/LatestUpdates";
import { Layout } from "~/components/Layout";

export type Props = React.ComponentProps<typeof LatestUpdates> & {
  deck: string;
  title: string;
};

export function Content({
  booklogUpdates,
  deck,
  movielogUpdates,
  title,
}: Props): JSX.Element {
  return (
    <Layout className="pb-8 bg-subtle">
      <SolidBackdrop
        deck={deck}
        title={title}
        titleStyle="text-2.5xl tablet:text-5xl font-bold uppercase tracking-widest desktop:text-7xl font-sans"
      />
      <LatestUpdates
        booklogUpdates={booklogUpdates}
        movielogUpdates={movielogUpdates}
      />
    </Layout>
  );
}
