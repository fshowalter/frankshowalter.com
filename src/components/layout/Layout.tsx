import type { JSX } from "react";

import { Backdrop } from "./Backdrop";
import { Footer } from "./Footer";
import { Logo } from "./Logo";

/**
 * Main layout component that provides the page structure with backdrop, header, and footer.
 * Wraps content in a consistent visual design across all pages.
 */
export function Layout({
  backdrop,
  children,
  className,
  logo = false,
}: {
  backdrop: React.ComponentProps<typeof Backdrop>;
  children: React.ReactNode;
  className?: string;
  logo?: boolean;
}): JSX.Element {
  return (
    <div className="h-full">
      <a
        className={`
          absolute top-0.5 left-1/2 z-50 mx-auto
          [transform:translate(-50%,calc(-100%-2px))]
          bg-subtle px-6 py-2 text-center text-accent
          focus:[transform:translate(-50%,0%)]
        `}
        href="#content"
      >
        Skip to content
      </a>
      <div className="flex min-h-full w-full flex-col bg-default">
        {logo && (
          <div
            className={`absolute top-0 left-0 z-30 px-container py-6 text-white`}
          >
            <Logo />
          </div>
        )}
        <main
          className={`
            z-10 grow
            ${className}
          `}
          id="content"
        >
          {backdrop && <Backdrop {...backdrop} />}
          {children}
        </main>
        <Footer />
      </div>
    </div>
  );
}
