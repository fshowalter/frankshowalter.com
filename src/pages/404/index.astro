---
import { Content } from "~/components/404/404";
import { getProps } from "~/components/404/getProps";
import Layout from "~/layouts/Layout.astro";

const title = "404: Not Found";
const deck = '"Dick Laurent is dead."';

const props = await getProps({
  deck,
  title,
});
---

<Layout
  canonical={false}
  meta={{ description: deck }}
  openGraph={false}
  title={title}
>
  <Content {...props} />
</Layout>
