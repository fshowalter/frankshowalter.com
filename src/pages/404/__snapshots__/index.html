<!doctype html>
<html class="h-full" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta content="width=device-width" name="viewport" />
    <link href="/favicon.svg" rel="icon" sizes="any" type="image/svg+xml" />
    <link href="/sitemap-index.xml" rel="sitemap" />
    <link
      as="font"
      crossorigin="anonymous"
      href="/fonts/Assistant/Assistant.woff2"
      rel="preload"
      type="font/woff2"
    />
    <link
      as="font"
      crossorigin="anonymous"
      href="/fonts/Frank-Ruhl-Libre/Frank-Ruhl-Libre_latin_300-900_normal.woff2"
      rel="preload"
      type="font/woff2"
    />
    <meta
      content="&#34;They're coming to get you, Barbra!&#34;"
      name="description"
    />
    <meta content="rgba(0,0,0,.85)" name="theme-color" />
    <title>404: Not Found - Frank Showalter</title>
  </head>
  <body class="h-full">
    <link
      rel="preload"
      as="image"
      imagesrcset="/_image/?href=%2Ffixtures%2Fassets%2Fbackdrops%2Fhome.png%3ForigWidth%3D2400%26origHeight%3D1350%26origFormat%3Dpng&amp;w=720&amp;h=405&amp;q=80&amp;f=avif 720w, /_image/?href=%2Ffixtures%2Fassets%2Fbackdrops%2Fhome.png%3ForigWidth%3D2400%26origHeight%3D1350%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=avif 1200w, /_image/?href=%2Ffixtures%2Fassets%2Fbackdrops%2Fhome.png%3ForigWidth%3D2400%26origHeight%3D1350%26origFormat%3Dpng&amp;w=1920&amp;h=1080&amp;q=80&amp;f=avif 1920w, /_image/?href=%2Ffixtures%2Fassets%2Fbackdrops%2Fhome.png%3ForigWidth%3D2400%26origHeight%3D1350%26origFormat%3Dpng&amp;w=2400&amp;h=1350&amp;q=80&amp;f=avif 2400w"
      imagesizes="100vw"
      fetchpriority="high"
    />
    <div class="h-full">
      <a
        class="absolute top-0.5 left-1/2 z-50 mx-auto [transform:translate(-50%,calc(-100%-2px))] bg-subtle px-6 py-2 text-center text-accent focus:[transform:translate(-50%,0%)]"
        href="#content"
        >Skip to content</a
      >
      <div class="flex min-h-full w-full flex-col bg-default">
        <main class="z-10 grow flex flex-col bg-subtle" id="content">
          <header
            class="relative flex min-h-[400px] w-full flex-col content-start items-center justify-end gap-6 bg-[#141414] pt-40 pb-8 text-white tablet:min-h-[640px] tablet:pt-40 tablet:pb-10 laptop:min-h-[clamp(640px,70vh,1350px)] laptop:pt-40 laptop:pb-16 desktop:pb-16"
          >
            <img
              class="absolute inset-0 size-full object-cover object-bottom"
              src="/_image/?href=%2Ffixtures%2Fassets%2Fbackdrops%2Fhome.png%3ForigWidth%3D2400%26origHeight%3D1350%26origFormat%3Dpng&amp;w=2400&amp;h=1350&amp;q=80&amp;f=avif"
              srcset="
                /_image/?href=%2Ffixtures%2Fassets%2Fbackdrops%2Fhome.png%3ForigWidth%3D2400%26origHeight%3D1350%26origFormat%3Dpng&amp;w=720&amp;h=405&amp;q=80&amp;f=avif    720w,
                /_image/?href=%2Ffixtures%2Fassets%2Fbackdrops%2Fhome.png%3ForigWidth%3D2400%26origHeight%3D1350%26origFormat%3Dpng&amp;w=1200&amp;h=675&amp;q=80&amp;f=avif  1200w,
                /_image/?href=%2Ffixtures%2Fassets%2Fbackdrops%2Fhome.png%3ForigWidth%3D2400%26origHeight%3D1350%26origFormat%3Dpng&amp;w=1920&amp;h=1080&amp;q=80&amp;f=avif 1920w,
                /_image/?href=%2Ffixtures%2Fassets%2Fbackdrops%2Fhome.png%3ForigWidth%3D2400%26origHeight%3D1350%26origFormat%3Dpng&amp;w=2400&amp;h=1350&amp;q=80&amp;f=avif 2400w
              "
              height="1350"
              sizes="100vw"
              width="2400"
              alt=""
              fetchpriority="high"
              loading="eager"
            />
            <div
              class="z-10 mx-auto flex w-full max-w-[908px] flex-col px-container after:absolute after:top-0 after:left-0 after:-z-10 after:h-full after:w-full after:bg-linear-to-t after:from-[rgba(0,0,0,.4)] after:via-transparent after:via-10% after:to-100% laptop:max-w-(--breakpoint-desktop)"
            >
              <h1
                class="text-[2rem] leading-10 font-extrabold [text-shadow:1px_1px_2px_#252525] tablet:text-4xl laptop:text-7xl"
              >
                404: Not Found
              </h1>
              <p
                class="mt-1 font-sans text-base [text-shadow:1px_1px_2px_black] laptop:my-4 laptop:text-xl"
              >
                &quot;They&#x27;re coming to get you, Barbra!&quot;
              </p>
            </div>
          </header>
          <div
            class="flex grow flex-col items-center justify-center px-container py-8 text-lg"
          >
            My apologies.
            <div class="mt-8 flex gap-4">
              <a
                class="relative font-semibold text-accent after:absolute after:bottom-0 after:left-0 after:h-px after:w-full after:origin-center after:scale-x-0 after:bg-accent after:transition-transform after:duration-500 hover:after:scale-x-100"
                href="/"
                >Home</a
              ><a
                class="relative font-semibold text-accent after:absolute after:bottom-0 after:left-0 after:h-px after:w-full after:origin-center after:scale-x-0 after:bg-accent after:transition-transform after:duration-500 hover:after:scale-x-100"
                href="https://www.franksmovielog.com/"
                >Movielog</a
              ><a
                class="relative font-semibold text-accent after:absolute after:bottom-0 after:left-0 after:h-px after:w-full after:origin-center after:scale-x-0 after:bg-accent after:transition-transform after:duration-500 hover:after:scale-x-100"
                href="https://www.franksbooklog.com/"
                >Booklog</a
              >
            </div>
          </div>
        </main>
        <div
          class="sticky bottom-0 flex w-full flex-wrap items-start justify-between gap-[10%] bg-footer px-container py-5 text-sm tablet:px-12 desktop:p-10"
        >
          <p
            class="w-full px-container text-center leading-5 font-normal text-inverse"
          >
            All images used in accordance with the<!-- -->
            <a
              class="relative inline-block pr-4 text-inherit underline underline-offset-3 hover:text-accent"
              href="http://www.copyright.gov/title17/92chap1.html#107"
              rel="nofollow"
              >Fair Use Law<svg
                class="absolute top-0.5 right-0 size-3"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5 6H5.25A2.25 2.25 0 0 0 3 8.25v10.5A2.25 2.25 0 0 0 5.25 21h10.5A2.25 2.25 0 0 0 18 18.75V10.5m-10.5 6L21 3m0 0h-5.25M21 3v5.25"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                ></path></svg></a
            >.
          </p>
        </div>
      </div>
    </div>
  </body>
</html>
